DataFormats/CTPPSDigi/src/CTPPSPixelDigiCollection.cc

```diff
@@ -0,0 +1,70 @@
+#include "DataFormats/CTPPSDigi/interface/CTPPSPixelDigiCollection.h"
+
+#include <algorithm>
+
+void CTPPSPixelDigiCollection::put(Range input, unsigned int detID) {
+/// put in Digis of detID
+
+/// store size of vector before put
+  IndexRange inputRange;
+
+/// put in CTPPSPixelDigis from input
+  bool first = true;
+
+/// fill input in temporary vector for sorting
+  std::vector<CTPPSPixelDigi> temporary;
+  auto sort_begin = input.first;
+  auto sort_end = input.second;
+  for ( ;sort_begin != sort_end; ++sort_begin ) {
+    temporary.push_back(*sort_begin);
```

Comment:
this can also be simplified substantially (and avoid all the push_backs at the same time)